setup.py

```diff
@@ -119,6 +133,22 @@ def package_version():
             'sklearn.js', 'sklearn-metadata.json',
             'tf.js', 'tf-metadata.json', 'tf-proto.js', 
             'tflite.js', 'tflite-metadata.json', 'tflite-schema.js', 
+
+            # OpenVINO IR
+            'openvino-ir/openvino-ir.js',
```

Comment:
Pattern so far is to have a single `xxxx.js` and `xxx-metadata.json` for each framework. This should be kept simple and easy to maintain. Try to fit it into a single `openvino.js` and `openvino-metadata.js` file like the other frameworks. Right now your approach seems more complex than TensorFlow or PyTorch.

Reply:
1. I have joined all the files in two: `openvino-ir.js`, `openvino-ir-parser.js` and refactored the codebase so that the logic is massively reused across the targets. I think that we are done with it.

2. I do not think that the early check is needed. It is already performed in the `open` function by throwing the exception if not readable and that exception is gracefuly handled by us

3. Agree, that is the desired improvement that should be done after the this PR is merged. I will try to do it for sure in my next PR.

@lutzroeder is this discussion resolved now?

Reply:
1. Getting close. Can you rename `openvino-ir.js` to just `openvino.js`, `openvino-ir-parser.js` and `openvino-parser.js` and `openvinoIR.ModelFactory` to `Openvino.ModelFactory`. You can use different namespace prefixes like `openvinoIR.Model` and `openvinoDot.Model` in the `openvino.js` file as they never need to be exported outside, only the unified `openvino.ModelFactory` is exported.

2. Basically any XML file the user opens that shows an error will be a bug on you? The unsupported check serves two purposes. a) filter out users loading random files from bugs b) if another framework supports XML files there might be another ModelFactory. If you can't handle a file it should be handed to the next factory. We can skip adding it for now but if usage will be any similar to JSON formats you will be dealing with this at some point.